components/datetime/src/fields/symbols.rs

```diff
@@ -38,6 +38,96 @@ pub enum FieldSymbol {
     TimeZone(TimeZone),
 }
 
+impl FieldSymbol {
+    /// Symbols are necessary components of `Pattern` struct, which
+    /// uses efficient byte serialization and deserialization via `zerovec`.
+    ///
+    /// The `FieldSymbol` impl provides non-public methods that can be used to efficiently
+    /// convert between `u8` and the symbol variant.
+    ///
+    /// The serialization model packages the variant in one byte.
+    ///
+    /// 1) The top four bits are used to determine the type of the field.
+    ///    (Examples: `Year`, `Month`, `Hour`)
+    ///
+    /// 2) The bottom four bits are used to determine the symbol of the type.
```

Comment:
thought (nb): perhaps mention something like "using that type's `from_idx()`/`idx()` for the mapping"